dbms/src/DataStreams/ExpandBlockInputStream.cpp

```diff
@@ -0,0 +1,54 @@
+// Copyright 2023 PingCAP, Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <DataStreams/ExpandBlockInputStream.h>
+#include <Interpreters/ExpressionActions.h>
+
+
+namespace DB
+{
+ExpandBlockInputStream::ExpandBlockInputStream(
+    const BlockInputStreamPtr & input,
+    const Expand2Ptr & expand2_,
+    const NamesAndTypes & names_and_types,
+    const String & req_id)
+    : expand2(expand2_)
+    , names_and_types(names_and_types)
+    , log(Logger::get(req_id))
+{
+    children.push_back(input);
+    i_th_project = 0;
+}
+
+Block ExpandBlockInputStream::getHeader() const
+{
+    Block res(names_and_types);
+    return res;
+}
```

Comment:
`ColumnPtr` is required for the Block returned by `getHeader`.
`ColumnPtr` is not needed only for `getSampleBlock`.

Reply:
like https://github.com/pingcap/tiflash/blob/1dc2d1af13e67f8c0b51414329aef71cbc5209ef/dbms/src/DataStreams/MockExchangeReceiverInputStream.h#L31-L33

Reply:
addressed, https://github.com/pingcap/tiflash/pull/7169/commits/2f2f3255e2f2a1b20572e086d3be9096292ac6fe